lib/resolve.go

```diff
@@ -0,0 +1,92 @@
+package lib
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/qri-io/dataset"
+	"github.com/qri-io/qri/base/dsfs"
+	"github.com/qri-io/qri/dsref"
+	"github.com/qri-io/qri/fsi"
+	"github.com/qri-io/qri/resolve"
+)
+
+// assert at compile time that instance is a RefResolver
+var _ resolve.RefResolver = (*Instance)(nil)
+
+// ParseAndResolveRef combines reference parsing and resolution
+func (inst *Instance) ParseAndResolveRef(ctx context.Context, refStr string) (dsref.Ref, error) {
+	ref, err := dsref.Parse(refStr)
+
+	// bad case references are allowed-but-warned for backwards compatibility
+	if errors.Is(err, dsref.ErrBadCaseName) {
```

Comment:
In the near future this should be a fatal error for every command *except* rename. Whether it's ignorable or not depends upon the caller, so it doesn't make sense to go this low in the stack.